core/liquidity/v2/snapshot.go

```diff
@@ -0,0 +1,461 @@
+package liquidity
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"code.vegaprotocol.io/vega/core/events"
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/num"
+	"code.vegaprotocol.io/vega/libs/proto"
+	"code.vegaprotocol.io/vega/logging"
+	typespb "code.vegaprotocol.io/vega/protos/vega"
+	snapshotpb "code.vegaprotocol.io/vega/protos/vega/snapshot/v1"
+)
+
+type SnapshotEngine struct {
+	*Engine
+
+	pl     types.Payload
+	market string
+
+	// liquidity types
+	stopped                bool
+	serialisedProvisions   []byte
+	serialisedPerformances []byte
+	serialisedSupplied     []byte
+	serialisedScores       []byte
+
+	// Keys need to be computed when the engine is instantiated as they are dynamic.
+	hashKeys        []string
+	provisionsKey   string
+	performancesKey string
+	scoresKey       string
+	suppliedKey     string
+}
+
+func (e *SnapshotEngine) Namespace() types.SnapshotNamespace {
+	return types.LiquidityV2Snapshot
+}
+
+func (e *SnapshotEngine) Keys() []string {
+	return e.hashKeys
+}
+
+func (e *SnapshotEngine) GetState(k string) ([]byte, []types.StateProvider, error) {
+	state, err := e.serialise(k)
+	return state, nil, err
+}
+
+func (e *SnapshotEngine) LoadState(ctx context.Context, p *types.Payload) ([]types.StateProvider, error) {
+	if e.Namespace() != p.Data.Namespace() {
+		return nil, types.ErrInvalidSnapshotNamespace
+	}
+
+	switch pl := p.Data.(type) {
+	case *types.PayloadLiquidityV2Provisions:
+		return nil, e.loadProvisions(ctx, pl.Provisions.GetLiquidityProvisions(), p)
+	case *types.PayloadLiquidityV2Performances:
+		return nil, e.loadPerformances(pl.Performances, p)
+	case *types.PayloadLiquidityV2Supplied:
+		return nil, e.loadSupplied(pl.Supplied, p)
+	case *types.PayloadLiquidityV2Scores:
+		return nil, e.loadScores(pl.Scores, p)
+	default:
+		return nil, types.ErrUnknownSnapshotType
+	}
+}
+
+func (e *SnapshotEngine) Stopped() bool {
+	return e.stopped
+}
+
+func (e *SnapshotEngine) StopSnapshots() {
+	e.log.Debug("market has been cleared, stopping snapshot production", logging.MarketID(e.marketID))
+	e.stopped = true
+}
+
+func (e *SnapshotEngine) OnMinProbabilityOfTradingLPOrdersUpdate(d num.Decimal) {
```

Comment:
given that the underlying engine is defined as *Engine I don't think you need all of those wrappers.

Reply:
That is true. I just stupidly copy / pasted the liquidity snapshot engine V1, thinking it's the way we do things.